src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -115,18 +172,22 @@ class WorkspaceNewRoomPage extends React.Component {
                         <View style={styles.mb5}>
                             <Text style={[styles.formLabel]}>{this.props.translate('newRoomPage.roomName')}</Text>
                             <RoomNameInput
-                                onChangeText={(roomName) => { this.setState({roomName}); }}
                                 initialValue={this.state.roomName}
                                 policyID={this.state.policyID}
+                                showErrorOnDemand
+                                errorText={this.state.errors.roomName}
+                                onChangeText={roomName => this.clearErrorAndSetValue('roomName', this.modifyRoomName(roomName))}
```

Comment:
Here, you take the error calculated on `RoomNameInput` and save it on state via `onChangeError`. 
And on submit, you will pick the error and move it to `this.state.errors.roomName` instead of calculating the same here. 
Rest would be the same as before. 

We do all this to keep the error logic on the `RoomNameInput` component and still show the error only onSubmit. But `RoomNameInput` is used somewhere else which shows the error on change of value.

Reply:
There could be other ways. Please thing about it.

Reply:
In this way we don't have access of blank room name error at first before changing the `roomName` state as the error validate will run on `RoomNameInput` only if there is change in room name state. I think this much is good by keeping  logic on both `RoomNameInput` and `WorkspaceNewRoomPage`.

Reply:
Why is there red outline on field when error is shown and not in other fields? There is no handling for that on `Picker`.

Reply:
@parasharrajat what can I do for it?

Reply:
there is `hasError` prop on the Picker which sets the outline.